packages/navigation/src/types.ts

```diff
@@ -0,0 +1,21 @@
+import type { NativeStackNavigationOptions } from '@react-navigation/native-stack';
+
+export interface WebScreenRule {
+  urlPattern: string;
+  title?: string;
+  presentation?: NativeStackNavigationOptions['presentation'];
+  component?: React.ElementType;
```

Comment:
I'm wondering if it should be the responsibility of the Web Screens to build the linking/routes for the paths with native screens (ie. a non-`VisitableView` component of the `Screen`) or instead that should be 'normal' React Navigation.

Reply:
A use case for the `component` option is to eg. a native screen on certain rourtes. We'd probably want the path/urlPattern to match a dynamic param, eg:

```js
{
   [Routes.Profile]: {
      urlPattern: "users/:id",
      component: ProfileScreen
   }
}
```

In this case, navigating to `/profile/mary` would render the `ProfileScreen` and we'd have `baseURL`, `path` *and* `id` available as params? I guess this could be as expected and working.

Reply:
If we want Web Screens to take ownership of generating the linking configuration for the whole NavigationContainer, including for routes which are native and not rendered as a webview, we should probably allow the options to match the linking options for React Navigation, ie. `parse`, `exact`, .. (`PathConfig`).

Alternatively we should think of a pattern where the Web Screen can be smoother 'mixed in' between a regular linking configuration and screen definition.